x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/AbstractBuilder.java

```diff
@@ -122,10 +122,18 @@ public static String unquoteString(Source source) {
             return null;
         }
 
-        // unescaped strings can be interpreted directly
+        // catch old method of ?" and ?' to define unescaped strings
         if (text.startsWith("?")) {
             checkForSingleQuotedString(source, text, 1);
```

Comment:
This is incorrect - in case of `?'` the error message is invalid - it tells that double quotes should be used which is not the case.
In fact I would argue that since the string is always quoted, if it starts with ? an exception should be thrown regardless of what follows it.

Reply:
That's true, It's a 2-step error for the user: `?'` -> `?"` -> `"""`. I just thought that explicitly handling the single quotes (no matter of the preceding `?`) was a "better" approach. I'll gladly change it.